keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,140 @@
+
+from keras_cv.layers import BaseImageAugmentationLayer
+from keras_cv.utils import preprocessing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomResizedCrop(BaseImageAugmentationLayer):
+    """
+    Randomly crops a part of an image and resizes it to provided size.
+
+    Args:
+        target_size: A uple of two integers used as the target size to crop
+            images to.
+        aspect_ratio_factor: (Optional) A tuple of two floats. Represents the
+            lower and upper bound for the aspect ratio of the cropped image
+            before resizing it to `target_size`. Defaults to (3./4., 4./3.).
+        area_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. Represents the lower and upper bound for
+            the area relative to the original image of the cropped image before
+            resizing it to `target_size`. Defaults to (0.08, 1.0).
+        interpolation: interpolation method used in the `ImageProjectiveTransformV3` op.
+             Supported values are `"nearest"` and `"bilinear"`.
+             Defaults to `"bilinear"`.
+        fill_mode: fill_mode in the `ImageProjectiveTransformV3` op.
+             Supported values are `"reflect"`, `"wrap"`, `"constant"`, and `"nearest"`.
+             Defaults to `"reflect"`.
+        fill_value: fill_value in the `ImageProjectiveTransformV3` op.
+             A `Tensor` of type `float32`. The value to be filled when fill_mode is
+             constant".  Defaults to `0.0`.
+        seed: Integer. Used to create a random seed.
+    """
+    def __init__(self,
+                 target_size,
+                 aspect_ratio_factor=(3. / 4., 4. / 3.),
+                 area_factor=(0.08, 1.0),
+                 interpolation="bilinear",
+                 fill_mode="reflect",
+                 fill_value=0.0,
+                 seed=None,
+                 **kwargs):
+        super(RandomResizedCrop, self).__init__(seed=seed, **kwargs)
+
+        self.target_size = target_size
+        self.aspect_ratio_factor = aspect_ratio_factor
+        self.area_factor = preprocessing.parse_factor(area_factor,
+                                                      param_name="area_factor",
+                                                      seed=seed)
+
+        self.interpolation = interpolation
+        self.fill_mode = fill_mode
+        self.fill_value = fill_value
+        self.seed = seed
+
+        if area_factor == 0.0 and aspect_ratio_factor == 0.0:
+            warnings.warn(
+                "RandomResizedCrop received both `area_factor=0.0` and "
+                "`aspect_ratio_factor=0.0`. As a result, the layer will perform no "
+                "augmentation.")
+
+    def get_random_transformation(self,
+                                  image=None,
+                                  label=None,
+                                  bounding_box=None):
+        area_factor = self.area_factor()
```

Comment:
In case `area_factor` was a float, then would `preprocessing.parse_factor()` ensure it gets converted to a function? If so, what kind?

Reply:
Yes. `preprocessing.parse_factor()` returns a `FactorSampler`: 
https://github.com/keras-team/keras-cv/blob/9da8e27bbfea0e860efc287006913e320b1aa4cb/keras_cv/utils/preprocessing.py#L117-L120

This object can be called later to get a random value in the specified range. However it does not support >1 max values at the moment.